### ISSUE TITLE
Stability issues relating to aiohttp timeouts during asyncpraw requests

### ISSUE THREAD

**Issue by a-rich** [a-rich/DJ-Tools#88]
**Describe the bug**
The Spotify playlist builder routinely fails when making requests to the Reddit API (see example stack trace below).
Despite the [30 second timeout](https://github.com/a-rich/DJ-Tools/blob/main/src/djtools/spotify/helpers.py#L195) on the `asyncpraw` client, the `asyncio.exceptions.TimeoutError` is raised almost immediately.
I'm not sure why this [inline exception handler](https://github.com/a-rich/DJ-Tools/blob/main/src/djtools/spotify/helpers.py#L267) (defined [here](https://github.com/a-rich/DJ-Tools/blob/main/src/djtools/utils/helpers.py)) isn't handling this `TimeoutError` either.

```
$ djtools --auto-playlist-update

2023-03-09 08:53:13 - djtools.spotify.helpers:274 - INFO - Filtering 100 "r/darkpsy" hot posts
Filtering 100 "r/darkpsy" hot posts: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 301314.94it/s]
2023-03-09 08:53:13 - djtools.spotify.helpers:307 - INFO - No new submissions from "r/darkpsy"
2023-03-09 08:53:13 - djtools.spotify.helpers:274 - INFO - Filtering 100 "r/psytrance" hot posts
Filtering 100 "r/psytrance" hot posts: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 1691251.61it/s]
2023-03-09 08:53:13 - djtools.spotify.helpers:307 - INFO - No new submissions from "r/psytrance"
2023-03-09 08:53:13 - djtools.spotify.helpers:369 - INFO - "r/Psytrance": https://open.spotify.com/playlist/4M5qAafmL65YkDaeFMjPH8
2023-03-09 08:53:13 - djtools.spotify.helpers:274 - INFO - Filtering 100 "r/minimaldnb" hot posts
Filtering 100 "r/minimaldnb" hot posts: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 1222829.15it/s]
2023-03-09 08:53:13 - djtools.spotify.helpers:307 - INFO - No new submissions from "r/minimaldnb"
2023-03-09 08:53:13 - djtools.spotify.helpers:274 - INFO - Filtering 100 "r/atmosphericdnb" hot posts
Filtering 100 "r/atmosphericdnb" hot posts: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 714532.20it/s]
2023-03-09 08:53:13 - djtools.spotify.helpers:307 - INFO - No new submissions from "r/atmosphericdnb"
2023-03-09 08:53:13 - djtools.spotify.helpers:274 - INFO - Filtering 100 "r/halftimednb" hot posts
Filtering 100 "r/halftimednb" hot posts: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 546133.33it/s]
2023-03-09 08:53:13 - djtools.spotify.helpers:307 - INFO - No new submissions from "r/halftimednb"
2023-03-09 08:53:13 - djtools.spotify.helpers:274 - INFO - Filtering 100 "r/spacebass" hot posts
Filtering 100 "r/spacebass" hot posts: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 991561.23it/s]
2023-03-09 08:53:13 - djtools.spotify.helpers:307 - INFO - No new submissions from "r/spacebass"
2023-03-09 08:53:14 - djtools.spotify.helpers:369 - INFO - "r/Minimaldnb": https://open.spotify.com/playlist/54sFYC28CDgd0QnAVkhs2A
2023-03-09 08:53:14 - djtools.spotify.helpers:369 - INFO - "Atmosphericdnb": https://open.spotify.com/playlist/2dAIZyeep7hblHhW007cN2
2023-03-09 08:53:14 - djtools.spotify.helpers:369 - INFO - "r/Spacebass": https://open.spotify.com/playlist/1X2o1tFyEp9AlidEoNNvTs
Traceback (most recent call last):
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/requestor.py", line 64, in request
    return await self._http.request(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/aiohttp/client.py", line 634, in _request
    break
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/aiohttp/helpers.py", line 721, in __exit__
    raise asyncio.TimeoutError from None
asyncio.exceptions.TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alrichards/.pyenv/versions/djtools/bin/djtools", line 8, in <module>
    sys.exit(dj_tools.main())
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/djtools/main.py", line 47, in main
    func(config)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/djtools/spotify/playlist_builder.py", line 184, in update_auto_playlists
    asyncio.run(async_update_auto_playlists(config))
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/djtools/spotify/playlist_builder.py", line 79, in async_update_auto_playlists
    tracks, subreddit = await task
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/asyncio/tasks.py", line 619, in _wait_for_one
    return f.result()  # May raise f.exception().
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/djtools/spotify/helpers.py", line 266, in get_subreddit_posts
    subs = [
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/djtools/spotify/helpers.py", line 266, in <listcomp>
    subs = [
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/models/listing/generator.py", line 63, in __anext__
    await self._next_batch()
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/models/listing/generator.py", line 89, in _next_batch
    self._listing = await self._reddit.get(self.url, params=self.params)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/util/deprecate_args.py", line 51, in wrapped
    return await _wrapper(*args, **kwargs)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/reddit.py", line 707, in get
    return await self._objectify_request(method="GET", params=params, path=path)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/reddit.py", line 815, in _objectify_request
    await self.request(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/util/deprecate_args.py", line 51, in wrapped
    return await _wrapper(*args, **kwargs)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncpraw/reddit.py", line 1032, in request
    return await self._core.request(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/sessions.py", line 370, in request
    return await self._request_with_retries(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/sessions.py", line 270, in _request_with_retries
    response, saved_exception = await self._make_request(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/sessions.py", line 187, in _make_request
    response = await self._rate_limiter.call(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/rate_limit.py", line 34, in call
    kwargs["headers"] = await set_header_callback()
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/sessions.py", line 322, in _set_header_callback
    await self._authorizer.refresh()
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/auth.py", line 371, in refresh
    await self._request_token(grant_type="client_credentials", **additional_kwargs)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/auth.py", line 153, in _request_token
    response = await self._authenticator._post(url, **data)
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/auth.py", line 33, in _post
    response = await self._requestor.request(
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/asyncprawcore/requestor.py", line 68, in request
    raise RequestException(exc, args, kwargs)
asyncprawcore.exceptions.RequestException: error with request
Exception ignored in: <function ClientSession.__del__ at 0x1094a2c10>
Traceback (most recent call last):
  File "/Users/alrichards/.pyenv/versions/djtools/lib/python3.8/site-packages/aiohttp/client.py", line 347, in __del__
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/asyncio/base_events.py", line 1740, in call_exception_handler
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 1475, in error
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 1589, in _log
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 1599, in handle
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 1661, in callHandlers
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 954, in handle
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 1186, in emit
  File "/Users/alrichards/.pyenv/versions/3.8.13/lib/python3.8/logging/__init__.py", line 1176, in _open
NameError: name 'open' is not defined

```

**To Reproduce**
NOTE: failure is more rare than success.
1. Configure DJ-Tools with the [appropriate options to run the Spotify playlist builder](https://github.com/a-rich/DJ-Tools/tree/main/src/djtools/spotify#playlist_builder).
2. Run DJ-Tools with the option that enables the Spotify playlist builder: `djtools --auto_playlist_update`

**Expected behavior**
Configured request timeouts should be honored -- exceptions after the configured timeout should be handled (log a warning) rather than being raised.

**Comment by a-rich** [comment]
2.4.1-b1 release contains this fix

**Comment by a-rich** [comment]
Closing as the issue is resolved in the [2.5.0](https://github.com/a-rich/DJ-Tools/releases/tag/v2.5.0) release.
